extensions/vscode-api-tests/src/singlefolder-tests/untitled.languagedetection.test.ts

```diff
@@ -0,0 +1,74 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as vscode from 'vscode';
+import { assertNoRpc, closeAllEditors } from '../utils';
+
+function sleep(ms: number) {
+	return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+suite('vscode - untitled automatic language detection', () => {
+
+	suiteSetup(async () => {
+		await vscode.workspace.getConfiguration()
+			.update('workbench.editor.untitled.experimentalLanguageDetection', true, vscode.ConfigurationTarget.Global);
+	});
+
+	suiteTeardown(async () => {
+		await vscode.workspace.getConfiguration()
+			.update('workbench.editor.untitled.experimentalLanguageDetection', undefined, vscode.ConfigurationTarget.Global);
+	});
+
+	teardown(async function () {
+		assertNoRpc();
+		await closeAllEditors();
+	});
+
+	test('test automatic language detection works', async () => {
+		const doc = await vscode.workspace.openTextDocument();
+		const editor = await vscode.window.showTextDocument(doc);
+
+		assert.strictEqual(editor.document.languageId, 'plaintext');
+		const result = await editor.edit(editBuilder => {
+			editBuilder.insert(new vscode.Position(0, 0), `{
+	"extends": "./tsconfig.base.json",
+	"compilerOptions": {
+		"removeComments": false,
+		"preserveConstEnums": true,
+		"sourceMap": false,
+		"outDir": "../out/vs",
+		"target": "es2020",
+		"types": [
+			"keytar",
+			"mocha",
+			"semver",
+			"sinon",
+			"winreg",
+			"trusted-types",
+			"wicg-file-system-access"
+		],
+		"plugins": [
+			{
+				"name": "tsec",
+				"exemptionConfig": "./tsec.exemptions.json"
+			}
+		]
+	},
+	"include": [
+		"./typings",
+		"./vs"
+	]
+}`);
+		});
+
+		assert.ok(result);
+
+		// language detection is debounced so we need to wait a bit
+		await sleep(2000);
```

Comment:
This is an integration test, but even there waiting 2s is ugly as it will make the builds slower.

Reply:
Yeah I don't know how to do this correctly then because I need to wait for the debounce to finish (600ms).

Any ideas? Also this test is failing even though it worked locally so I'm guessing this sleep is the cause.

Reply:
I was originally thinking there would be some kind of event that I could set up but that doesn't seem to be the case...unless I'm wrong.

Reply:
Figured it out.